Formula/gstreamer.rb

```diff
@@ -1,71 +1,262 @@
+  # https://gitlab.freedesktop.org/gstreamer/gst-python/-/merge_requests/41
+  # TODO: Migrate patch to gstreamer monorepo.
+  patch :DATA
 
   def install
-    # Ban trying to chown to root.
-    # https://bugzilla.gnome.org/show_bug.cgi?id=750367
-    args = std_meson_args + %w[
+    (buildpath/"subprojects/gst-plugins-rs").install resource("rs")
+    site_packages = Language::Python.site_packages(python3)
+    # To pass arguments to subprojects (e.g. `gst-editing-services`), use
+    #   -Dsubproject:option=value
+    args = %W[
+      -Dpython=enabled
+      -Dlibav=enabled
+      -Dlibnice=disabled
+      -Dbase=enabled
+      -Dgood=enabled
+      -Dugly=enabled
+      -Dbad=enabled
+      -Ddevtools=enabled
+      -Dges=enabled
+      -Drtsp_server=enabled
+      -Drs=enabled
+      -Dtls=enabled
+      -Dqt5=disabled
+      -Dtools=enabled
+      -Dorc-source=system
+      -Dgpl=enabled
+      -Dtests=disabled
+      -Dexamples=disabled
+      -Dnls=enabled
+      -Dorc=enabled
+      -Ddoc=disabled
+      -Dgtk_doc=disabled
       -Dintrospection=enabled
-      -Dptp-helper-permissions=none
+      -Dpackage-origin=#{tap.default_remote}
+      -Dgst-devtools:validate=enabled
+      -Dgst-devtools:cairo=enabled
+      -Dgst-editing-services:pygi-overrides-dir=#{site_packages}/gi/overrides
+      -Dgst-python:pygi-overrides-dir=#{site_packages}/gi/overrides
+      -Dgst-python:python=#{python3}
+      -Dgst-plugins-bad:sctp=enabled
+      -Dgst-plugins-bad:sctp-internal-usrsctp=disabled
+      -Dgst-plugins-good:soup=enabled
+      -Dgst-plugins-rs:closedcaption=enabled
+      -Dgst-plugins-rs:dav1d=enabled
+      -Dgst-plugins-rs:sodium=enabled
+      -Dgst-plugins-rs:csound=disabled
+      -Dgst-plugins-rs:gtk4=enabled
+      -Dgst-plugins-rs:sodium-source=system
     ]
 
-    # Look for plugins in HOMEBREW_PREFIX/lib/gstreamer-1.0 instead of
-    # HOMEBREW_PREFIX/Cellar/gstreamer/1.0/lib/gstreamer-1.0, so we'll find
-    # plugins installed by other packages without setting GST_PLUGIN_PATH in
-    # the environment.
-    inreplace "meson.build",
-      "cdata.set_quoted('PLUGINDIR', join_paths(get_option('prefix'), get_option('libdir'), 'gstreamer-1.0'))",
-      "cdata.set_quoted('PLUGINDIR', '#{HOMEBREW_PREFIX}/lib/gstreamer-1.0')"
-
-    mkdir "build" do
-      system "meson", *args, ".."
-      system "ninja", "-v"
-      system "ninja", "install", "-v"
-    end
+    # The apple media plug-in uses API that was added in Mojave
+    args << "-Dgst-plugins-bad:applemedia=disabled" if MacOS.version <= :high_sierra
+
+    # Ban trying to chown to root.
+    # https://bugzilla.gnome.org/show_bug.cgi?id=750367
+    args << "-Dgstreamer:ptp-helper-permissions=none"
 
-    bin.env_script_all_files libexec/"bin", GST_PLUGIN_SYSTEM_PATH: HOMEBREW_PREFIX/"lib/gstreamer-1.0"
+    # Prevent the build from downloading an x86-64 version of bison.
+    inreplace "meson.build", "subproject('macos-bison-binary')", ""
```

Comment:
Ah, hmm, we should provide a way to disable this. I'll fix it upstream :)

Reply:
https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4428

You can start using this with the 1.24 release.

Reply:
Thanks!